sentry_logging/lib/src/logging_integration.dart

```diff
@@ -0,0 +1,76 @@
+import 'dart:async';
+
+import 'package:logging/logging.dart';
+import 'package:sentry/sentry.dart';
+import 'version.dart';
+import 'extension.dart';
+
+/// An [Integration] which listens to all messages of the
+/// [logging](https://pub.dev/packages/logging) package.
+class LoggingIntegration extends Integration<SentryOptions> {
+  /// Creates the [LoggingIntegration].
+  ///
+  /// Setting [logExceptionAsEvent] to true (default) captures all
+  /// messages with errors as an [SentryEvent] instead of an [Breadcrumb].
+  /// Setting [logExceptionAsEvent] to false captures everything as
+  /// [Breadcrumb]s.
+  LoggingIntegration({
+    Level minBreadcrumbLevel = Level.INFO,
+    Level minEventLevel = Level.SEVERE,
+  })  : _minBreadcrumbLevel = minBreadcrumbLevel,
+        _minEventLevel = minEventLevel;
+
+  final Level _minBreadcrumbLevel;
+  final Level _minEventLevel;
+  late StreamSubscription<LogRecord> _subscription;
+  late Hub _hub;
+
+  @override
+  FutureOr<void> call(Hub hub, SentryOptions options) {
+    _hub = hub;
+    _setSdkVersion(options);
+    _subscription = Logger.root.onRecord.listen(
+      _onLog,
+      onError: (Object error, StackTrace stackTrace) {
+        _hub.captureException(error, stackTrace: stackTrace);
+      },
+    );
+    options.sdk.addIntegration('LoggingIntegration');
+  }
+
+  @override
+  Future<void> close() async {
+    await super.close();
+    await _subscription.cancel();
+  }
+
+  void _setSdkVersion(SentryOptions options) {
+    final sdk = SdkVersion(
+      name: sdkName,
+      version: sdkVersion,
+      integrations: options.sdk.integrations,
+      packages: options.sdk.packages,
+    );
+    sdk.addPackage('pub:sentry_logging', sdkVersion);
+    options.sdk = sdk;
```

Comment:
Should this really replace the current SDK? This seems to be quite much because this ist just added as an integration.

Reply:
no, the idea is that you merge the packages.
https://github.com/getsentry/sentry-java/blob/main/sentry-android-timber/src/main/java/io/sentry/android/timber/SentryTimberIntegration.kt#L45-L53
See what `SdkVersion.updateSdkVersion` does.
https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/protocol/SdkVersion.java#L124-L144

Reply:
Wouldn't it be then enough to just add the package?

Reply:
the difference is that the `sdk` object is the reporter and the one highlighted in the UI, the package list is an additional, so you know which package actually "captured" the event

Reply:
I think it's already like in the code you referenced. I have to assign the `sdk` because it's immutable.

Reply:
indeed, missed the part that does, `packages: options.sdk.packages`